hudi-common/src/main/java/org/apache/hudi/common/table/read/HoodiePositionBasedFileGroupRecordBuffer.java

```diff
@@ -84,6 +79,10 @@ public void processDataBlock(HoodieDataBlock dataBlock, Option<KeySpec> keySpecO
       isFullKey = keySpecOpt.get().isFullKey();
     }
 
+    if (dataBlock.containsPartialUpdates()) {
+      enablePartialMerging = true;
+    }
```

Comment:
Is there any chance this flag switch back to false? Can the buffer consumes multiple log blocks?

Reply:
The buffer consumes multiple log blocks in the file group.  When there are partial updates in one log block, we switch the merging mode and do not switch back going forward to guarantee correct merging results.